api/proxy/servers/arbitrum_altda/handlers.go

```diff
@@ -0,0 +1,186 @@
+// failover error message if disableFallbackStoreDataOnChain=true
+//
+// TODO: Determine the encoding standard to use for the returned DA Commitment. It's assumed that an EigenDAV2 message
+// header byte will be prefixed. We can likely reuse the Standard Commitment mode but will require some analysis.
+//
+// TODO: Add processing for client provided timeout value
+func (s *Server) Store(
+	ctx context.Context,
+	message hexutil.Bytes,
+	timeout hexutil.Uint64,
+	disableFallbackStoreDataOnChain bool,
+) (StoreResult, error) {
+	panic("Store is unimplemented")
+}
+
+// Generate proof is used to prove a 32 byte CustomDA preimage type for READPREIMAGE
+// The exact implementation here is still a bit TBD - but we'll prove availability of the 32 bytes
+// by computing a kzg point opening proof using the data commitment provided in the DA Cert.
+// This will be equivalent to what's already done in the arbitrator for serializing an EigenDA READPREIMAGE
+// proof. The large difference is this is done on the Custom DA server in go code as an
+// "extension" of the one step proof
+// construction logic.
+//
+// READPREIMAGE only cares about the availability or corectness of an EigenDA blob wrt it's kzg data commitment that's
+// persisted in the already agreed upon DA Cert.
+// Let's assumes that the EigenDA disperser would never sign over a DA Cert with an invalid data commitment.
+// Pulling that off would require majority corruption of the EigenDA operator quorums and collusion with disperser
+// which is a highly improbable event.
+// The data commitment is a tamper resistant field in the rollup domain since modification would result
+// in an incorrect merkle leaf hash being constructed from the blob header and result in an invalid merkle inclusion
+// proof which would be treated as an invalid DA Cert by the rollup.
+//
+// TODO: Generating the data witness "opening" proof requires access to the entire EigenDA blob
+// which isn't provided by client here. We can do a storage retrieval operation through the EigenDA Manager
+// to fetch the blob corresponding to the DA Cert. Redundantly performing DA Cert verification is a necessary
+// invariant here to strictly enforce given that this function would only ever be called if checkDACert(DA Cert)=true.
+// It's slow to do another storage lookup but performance considerations are irrelevant given this is only callable
+// in the worst case one step proof.
+//
+// TODO: Determine encoding standard that's also understood for onchain verification
+//
+// // current encoding proposal:
+// // // SERIALIZED EigenDA V2 CUSTOM READ PREIMAGE PROOF:
+// // // kzg commitment and preimage length are extractable
+// // // from the existing DA Cert
+//
+// // // [0:32]  - root of unity @ field element offset
+// // // [32:64] - field element or preimageChunk being one step proven
+// // // [64:128] - point opening proof (g1 point)
+// // // [128:256] - g2TauMinusG2z
```

Comment:
nit: reads a bit weird with all the //

Reply:
a21569eeba4021d16cda6954ef25809eecc77974